lambdas/url_access_patterns/lambda_for_updateURL/index.js

```diff
@@ -1,35 +1,35 @@
 const AWS = require("aws-sdk");
-const dynamo = new AWS.DynamoDB.DocumentClient();+
-
-function response(statusCode, message) {
+const dynamo = new AWS.DynamoDB.DocumentClient();
++function response(statusCode, message) {
   return {
     statusCode: statusCode,
     body: message,
   };
-}
+};
 
-exports.updateURL = async (event) =>{
-    const reqBody = event.body;
-    var params = {
-        TableName:"V-Transfer",
-        Key: {
-            PK: `FILE#${event.path.f_timestamp}`,
-            SK: `URL#${reqBody.u_timestamp}`,
-        },
-        ReturnValues:"ALL_NEW"
-    }
-    if(reqBody.visible!=undefined){    //request to change  visiblity of url
-        params.UpdateExpression="set visible = :val"
-        params.ExpressionAttributeValues= {":val":reqBody.visible}
-    }
-    else if(reqBody.clicks_left!=undefined){   //request to update clicks_left for url
-        params.UpdateExpression="set clicks_left = :val"
-        params.ExpressionAttributeValues= {":val":reqBody.clicks_left}
-    }
-    try{
-        const updated_url_data=  await dynamo.update(params).promise();
-        return response(200,updated_url_data.Attributes);
-    }catch(err){
-        return response(err.statusCode,err.message)
-    }
-}
\ No newline at end of file
+exports.handler = async (event) => {
+  const reqBody = event.body;
+  var params = {
+    TableName: "V-Transfer",
+    Key: {
+      PK: `FILE#${event.path.fileId}`,
+      SK: `URL#${event.path.urlId}`,
+    },
+    ReturnValues: "ALL_NEW",
+  };
+  if (reqBody.visible != undefined) {
+    //request to change  visiblity of url
+    params.UpdateExpression = "set visible = :val";
+    params.ExpressionAttributeValues = { ":val": reqBody.visible };
+  } else if (reqBody.clicks_left != undefined) {
+    //request to update clicks_left for url
+    params.UpdateExpression = "set clicks_left = :val";
+    params.ExpressionAttributeValues = { ":val": reqBody.clicks_left };
+  }
```

Comment:
what is user want to update both the feature, i.e. he/she wants to `set the visibility to true and the `click count to 10` ??

Since the code is structured in if-else if , only one feature would be updated once.

Reply:
We would not be providing the feature to update `cilcks_left` to a user that can only be incremented through the admin portal after the user sends a request to update so due to certain restrictions like user type or due to a limit of the number of ipfs requests.

Reply:
no one told me about this feature, 
fine if its so